crates/iceberg/src/spec/values.rs

```diff
@@ -966,6 +978,547 @@ mod timestamptz {
     }
 }
 
+mod serde {
```

Comment:
I would suggest to move this into another pr and resolve them first, otherwise this pr would get too large to review.

Reply:
If `RawLiteral` looks well, I can seperate it to another PR.

Reply:
I'm ok with this approach, cc @JanKaul @Xuanwo

Reply:
I think using the RawLiteral makes sense. Initially I tried to make `Literal` directly deserializable from avro but the types have to be converted from avro types to iceberg types anyway.